### PR TITLE
OCM-201: Add support for mapping_method

### PR DESCRIPTION
**Pull request by sagidayan**
Added mapping_method support.
 - Validation
 - idp create
 - idp read
 - subsytem tests

### CODE REVIEW

**Comment by gdbranco** [comment]
Considering ROSA CLI flow SGTM

File: provider/identity_provider_resource.go [hunk 1]
```diff
@@ -59,6 +62,36 @@ func (t *IdentityProviderResourceType) GetSchema(ctx context.Context) (result tf
...
 				Required:    true,
 			},
+			"mapping_method": {
+				Description: "Specifies how new identities are mapped to users when they log in. Options are [add claim generate lookup] (default 'claim')",
+				Type:        types.StringType,
+				Optional:    true,
+				Validators: []tfsdk.AttributeValidator{
+					&common.AttributeValidator{
+						Desc: "Validate mapping_method",
+						Validator: func(ctx context.Context, req tfsdk.ValidateAttributeRequest, resp *tfsdk.ValidateAttributeResponse) {
```
Comment (bardielle) [inline provider/identity_provider_resource.go:72]: I think it's better to move that function into idp package....

File: provider/identity_provider_resource.go [hunk 2]
```diff
@@ -139,10 +172,16 @@ func (r *IdentityProviderResource) Create(ctx context.Context,
 		)
 		return
 	}
+	// handle mapping_method
+	if state.MappingMethod.Unknown || state.MappingMethod.Null {
+		// Set default Value
+		state.MappingMethod.Value = defaultMappingMethod
```
Comment (bardielle) [inline provider/identity_provider_resource.go:178]: you can't do it
this is an optional only attribute
If you will set it here you change the value of that attribute.... you have 2 options: 
 1. change it to be also computed
 2. no setting the value if it wasn't set

File: provider/identity_provider_resource.go [hunk 3]
```diff
@@ -270,6 +309,10 @@ func (r *IdentityProviderResource) Read(ctx context.Context, request tfsdk.ReadR
 	state.Name = types.String{
 		Value: object.Name(),
 	}
+	state.MappingMethod = types.String{
```
Comment (bardielle) [inline provider/identity_provider_resource.go:312]: you can't do it (the same comment as before....)

**Comment by oriAdler** [comment]
LGTM

**Pull request merged**